### PR TITLE
add export properties buyers to csv

### LINKED ISSUE #60: As a User, I can export properties/buyers to csv, so that I can transfer PropertyWhiz data into spreadsheets

### PR DESCRIPTION
**Pull request by guangxuan**
Exported properties look like this
[properties2.csv](https://github.com/AY2122S1-CS2103T-W11-4/tp/files/7330424/properties2.csv)

Exported buyers look like this
[buyers.csv](https://github.com/AY2122S1-CS2103T-W11-4/tp/files/7334689/buyers.csv)

Only adds menu functions for exporting, does not add commands yet.

closes #60

### CODE REVIEW

File: src/main/java/seedu/address/storage/CsvManager.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.property.Buyer;
+import seedu.address.model.property.Property;
+
+/**
+ * Manages exporting to or importing from csv.
+ */
+public class CsvManager {
```
Comment (waynetee) [inline src/main/java/seedu/address/storage/CsvManager.java:21]: should everything here be static?

**Review by waynetee — COMMENTED** [review]
Nice job overall, added my comments

File: src/main/java/seedu/address/storage/CsvManager.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private final Logger logger = LogsCenter.getLogger(CsvManager.class);
+
+    /**
+     * Exports properties to csv file.
+     */
+    public void exportProperties(File file, ObservableList<Property> properties) {
```
[the remark this answers is not in the document]
Comment (guangxuan) [reply src/main/java/seedu/address/storage/CsvManager.java:27]: List is better here yes.

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 3]
```diff
@@ -28,6 +30,7 @@
     private final Model model;
     private final Storage storage;
     private final AddressBookParser addressBookParser;
+    private final CsvManager csvManager;
```
[the remark this answers is not in the document]
Comment (guangxuan) [reply src/main/java/seedu/address/logic/LogicManager.java:33]: Now it's part of Storage.

File: src/main/java/seedu/address/storage/CsvManager.java (hunk 1 above)
Comment (guangxuan) [reply src/main/java/seedu/address/storage/CsvManager.java]: mmm CsvManager is now in Storage, and storage does not seem to use static.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -154,6 +158,43 @@ public void handleHelp() {
         }
     }
 
+    /**
+     * Gets user to select a destination for saving csv.
+     *
+     * @param title Title of fileChooser dialog box
+     * @return File object chosen by user
```
Comment (waynetee) [inline src/main/java/seedu/address/ui/MainWindow.java:165]: I think it is worth documenting that it is null when user cancels

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 5]
```diff
@@ -76,4 +86,42 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) thro
         addressBookStorage.saveAddressBook(addressBook, filePath);
     }
 
+    // ================ Import/ Export Csv methods ==============================
+
+    @Override
+    public void exportProperties(ReadOnlyAddressBook addressBook, File file) throws IOException {
```
Comment (waynetee) [inline src/main/java/seedu/address/storage/StorageManager.java:92]: These methods can go in a separate class to improve code quality (srp/slap/facade pattern)

**Review by waynetee — APPROVED** [review]
LGTM, remember to add error handling (feedback to user) in next pr

**Pull request merged**